userspace/libsinsp/container.cpp

```diff
@@ -551,53 +553,74 @@ void sinsp_container_manager::create_engines() {
 		                                                                   m_static_name,
 		                                                                   m_static_image);
 		m_container_engines.push_back(engine);
-		m_container_engine_by_type[CT_STATIC] = engine;
+		m_container_engine_by_type[CT_STATIC].push_back(engine);
 		return;
 	}
 #if !defined(MINIMAL_BUILD) && !defined(__EMSCRIPTEN__)
 #ifndef _WIN32
 	if(m_container_engine_mask & (1 << CT_PODMAN)) {
 		auto podman_engine = std::make_shared<container_engine::podman>(*this);
 		m_container_engines.push_back(podman_engine);
-		m_container_engine_by_type[CT_PODMAN] = podman_engine;
+		m_container_engine_by_type[CT_PODMAN].push_back(podman_engine);
 	}
 	if(m_container_engine_mask & (1 << CT_DOCKER)) {
 		auto docker_engine = std::make_shared<container_engine::docker_linux>(*this);
 		m_container_engines.push_back(docker_engine);
-		m_container_engine_by_type[CT_DOCKER] = docker_engine;
+		m_container_engine_by_type[CT_DOCKER].push_back(docker_engine);
 	}
 
 	if(m_container_engine_mask & ((1 << CT_CRI) | (1 << CT_CRIO) | (1 << CT_CONTAINERD))) {
-		auto cri_engine = std::make_shared<container_engine::cri>(*this);
-		m_container_engines.push_back(cri_engine);
-		m_container_engine_by_type[CT_CRI] = cri_engine;
-		m_container_engine_by_type[CT_CRIO] = cri_engine;
-		m_container_engine_by_type[CT_CONTAINERD] = cri_engine;
+		// Get CRI socket paths from settings
+		libsinsp::cri::cri_settings& cri_settings = libsinsp::cri::cri_settings::get();
+		if(cri_settings.get_cri_unix_socket_paths().empty()) {
+			// Add default paths
+			cri_settings.add_cri_unix_socket_path("/run/containerd/containerd.sock");
+			cri_settings.add_cri_unix_socket_path("/run/crio/crio.sock");
+			cri_settings.add_cri_unix_socket_path("/run/k3s/containerd/containerd.sock");
+		}
+
+		const auto& cri_socket_paths = cri_settings.get_cri_unix_socket_paths();
+
+		size_t engine_index = 0;
+		for(auto socket_path : cri_socket_paths) {
+			auto cri_engine =
+			        std::make_shared<container_engine::cri>(*this, socket_path, engine_index);
+			m_container_engines.push_back(cri_engine);
+			m_container_engine_by_type[CT_CRI].push_back(cri_engine);
+			m_container_engine_by_type[CT_CRIO].push_back(cri_engine);
+			m_container_engine_by_type[CT_CONTAINERD].push_back(cri_engine);
```

Comment:
Do we still want to add the CRI sockets as containerd sockets?

Reply:
Good point. I don't see why we should keep it the way it is now.